pkg/flow/memory.go

```diff
@@ -165,6 +165,24 @@ func (im *instanceMemory) StoreData(key string, val interface{}) error {
 
 }
 
+func (im *instanceMemory) tags() map[string]string {
+	tag := make(map[string]string)
+	if im.logic == nil {
+		return nil
+	}
+	tag["ID"] = im.logic.GetID()
+	if im.logic == nil {
+		tag["nil"] = "nil" //TODO: handle this case
```

Comment:
This can't happen because the previous identical if condition returns. But if this did happen, can you explain what the goal is?

Reply:
you are right this is redundant code.
i removed it